src/api/controllers/authController.js

```diff
@@ -0,0 +1,65 @@
+import bcrypt, { hashSync, genSaltSync } from 'bcrypt';
+import models from '../models/index';
+import generateToken from '../../helpers/tokens/generate.token';
+import sendResult from '../../helpers/results/send.auth';
+
+const { User } = models;
+
+/**
+ *
+ *
+ * @export
+ * @class Auth
+ */
+export default class Auth {
+  /**
+   *
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns
+   * @memberof Auth
+   * @returns {Boolean} true
+   */
+  static async signup(req, res) {
+    const { names, email, password } = req.body;
+    const salt = genSaltSync(parseFloat(process.env.BCRYPT_HASH_ROUNDS) || 10);
+    const hashedPassword = hashSync(password, salt);
+    const user = await User.create({
+      names,
+      email,
+      password: hashedPassword,
+    });
+    const token = generateToken(user);
+    return sendResult(res, 201, 'user created successfully', user, token);
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @memberof Auth
+   * @returns {Boolean} true
+   */
+  static login(req, res) {
+    const { email, password } = req.body;
+    User.findByEmail(email).then((user) => {
+      if (user) {
+        const isPasswordValid = bcrypt.compareSync(password, user.dataValues.password);
+        if (isPasswordValid) {
+          const token = generateToken(user);
+          return sendResult(res, 200, 'user created successfully', user, token);
```

Comment:
@alainburindi I think this should be a different message. Perhaps, 'user logged in successfully'

Reply:
done